boringtun/src/noise/handshake.rs

```diff
@@ -30,25 +32,22 @@ static INITIAL_CHAIN_HASH: [u8; KEY_LEN] = [
 
 macro_rules! HASH {
     ($data1:expr, $data2:expr) => {
-        Blake2s::new_hash().hash(&$data1).hash(&$data2).finalize()
```

Comment:
Can we get rid of these macros? We should not be mixing macros and cryptography.

Reply:
Not easily, no, and certainly not in this PR. These are in place of the KDF lines in the specification to make it easier to read and audit. Surprisingly, they do actually make sense here. The problem is the use of `&` inside the macro, which does harm readability.